scripts/scraper.py

```
@@ -17,39 +17,115 @@
-            if response.status_code == 404:
+            response = SESSION.get(url, timeout=DEFAULT_TIMEOUT)
+            if response.status_code == requests.codes.not_found:
                 raise NotFound(url)
             response.raise_for_status()
             return BeautifulSoup(response.text, "html.parser")
         except requests.RequestException:
             if attempt == MAX_RETRIES - 1:
                 raise
-            time.sleep(1)
-
+            time.sleep(RATE_LIMIT_DELAY)
```

Ruff (0.16.1)

[error] Local variable `RATE_LIMIT_DELAY` referenced before assignment

(F823)